great_expectations/expectations/core/expect_query_results_to_match_source.py

```diff
@@ -296,6 +299,110 @@ def _validate(
                 },
             }
 
+    @override
+    @classmethod
+    @renderer(renderer_type=AtomicDiagnosticRendererType.OBSERVED_VALUE)
+    def _atomic_diagnostic_observed_value(
+        cls,
+        configuration: Optional[ExpectationConfiguration] = None,
+        result: Optional[ExpectationValidationResult] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> list[RenderedAtomicContent]:
+        details = cls._get_details_from_results(result)
+
+        missing_rows: list[dict[str, Any]] = details["missing_rows"]
+        unexpected_rows: list[dict[str, Any]] = details["unexpected_rows"]
+        missing_rows_table = cls._create_observed_values_table(missing_rows)
+        unexpected_rows_table = cls._create_observed_values_table(unexpected_rows)
+
+        return [
+            cls._create_table_rendered_atomic_content(
+                unexpected_rows_table,
+                label="Unexpected records",
+            ),
+            cls._create_table_rendered_atomic_content(
+                missing_rows_table,
+                label="Missing records",
+            ),
+        ]
+
+    @classmethod
+    def _get_details_from_results(
+        cls, result: Optional[ExpectationValidationResult]
+    ) -> dict[str, Any]:
+        assert result and result.result, "Must have result"
+        details = result.result.get("details")
+        assert isinstance(details, dict), "Details must exist and be a dict"
+        return details
+
+    @classmethod
+    def _create_table_rendered_atomic_content(
+        cls,
+        table: list[list[RendererTableValue]],
+        label: str,
+    ) -> RenderedAtomicContent:
+        rows = table[1:]
+        template = f"{label}: $row_count"
+        params = {
+            "row_count": {
+                "schema": RendererSchema(type=RendererValueType.NUMBER),
+                "value": len(rows),
+            }
+        }
+
+        if not table:
+            return RenderedAtomicContent(
+                name=AtomicDiagnosticRendererType.OBSERVED_VALUE,
+                value=RenderedAtomicValue(
+                    schema={"type": "StringValueType"},
+                    template=template,
+                    params=params,
+                ),
+                value_type="StringValueType",
```

Comment:
I think this should still be `TableType`, because anything with `StringValueType` will get the vertical red or green bar next to it (as if it is the prescriptive string/description).

Reply:
Do we have this documented somewhere? I'm guessing it's tribal knowledge? I've felt pretty blind working on these PRs